pkg/gateway/handler/cloudcode.go

```diff
@@ -0,0 +1,33 @@
+package handler
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"github.com/skygeario/skygear-server/pkg/gateway/config"
+	"github.com/skygeario/skygear-server/pkg/gateway/model"
+)
+
+func NewCloudCodeHandler(routerConfig config.RouterConfig) http.HandlerFunc {
+	proxy := newCloudCodeReverseProxy(routerConfig)
+	return proxy.ServeHTTP
+}
+
+func newCloudCodeReverseProxy(routerConfig config.RouterConfig) *httputil.ReverseProxy {
+	director := func(req *http.Request) {
+		query := req.URL.RawQuery
+		fragment := req.URL.Fragment
+		var err error
+		u, err := url.Parse(routerConfig.CloudCodeGatewayURL)
+		if err != nil {
+			panic(err)
+		}
+		cloudCode := model.CloudCodeFromContext(req.Context())
+		req.URL = u
+		req.URL.Path = "/function" + cloudCode.TargetPath
```

Comment:
@Steven-Chan Just correct me if I misunderstand anything :)

1. `/function` here means the type of cloud function? and we will have something like `/http-handler` in the future?
2. Forward path seems missing when forward to cloud function gateway?